lib/teleterm/clusters/storage.go

```diff
@@ -161,19 +155,27 @@ func (s *Storage) addCluster(ctx context.Context, dir, webProxyAddress string) (
 }
 
 // fromProfile creates a new cluster from its profile
-func (s *Storage) fromProfile(clusterName string) (*Cluster, error) {
-	if clusterName == "" {
+func (s *Storage) fromProfile(profileName, leafClusterName string) (*Cluster, error) {
```

Comment:
There's an outstanding issue in Teleport Connect where all our URIs use the proxy hostname as the cluster name (which is equivalent to a profile name, I think). I'll create a separate ticket for it soon, but in general the problem is that the proxy hostname can be different than the root cluster name.

So when `Storage.GetByURI` calls `fromProfile`, this name is actually the profile name and not the real root cluster name from the key. I changed the argument name here from `clusterName` to `profileName` to signal that for the future.